operator/controllers/common/phasehandler.go

```diff
@@ -79,6 +81,11 @@ func (r PhaseHandler) HandlePhase(ctx context.Context, ctxTrace context.Context,
 			piWrapper.Complete()
 			piWrapper.SetState(common.StateFailed)
 			spanTrace.AddEvent(phase.LongName + " has failed")
+			if msg, err := r.createFailureReasonSpanEvents(ctx, phase, piWrapper); err != nil {
+				r.Log.Error(err, "cannot create failure spans")
+			} else {
+				spanTrace.AddEvent(msg)
```

Comment:
Maybe would be better to add one event per task/eval at the timestamp they were computed (if we have this info). 
You can use `EventOption` and its `timestamp` field for this: https://github.com/open-telemetry/opentelemetry-go/blob/main/trace/config.go#L147

Reply:
we are basically adding this information after all evaluations/tasks are finished, it does not make much sense to add them with timestamps, but if preferred, we can add an event per failed evaluation/task, would you prefer this ?

Reply:
done as multiple events -> event per failure